osamuaoki/incus-ui-canonical
# Merge lxd-ui v0.8

Merge [lxd 0.8](https://github.com/canonical/lxd-ui/tree/0.8) into debian branch. Existing codebase is on top of 0.6, comparison between v0.6 to debian branch: https://github.com/osamuaoki/incus-ui-canonical/compare/main...debian

---

Merge conflicts on these files:

- src/pages/images/ImageSelector.tsx
- src/pages/login/Login.tsx
- src/pages/storage/UploadCustomImageHint.tsx

---

Test steps:

1. `apt install yarnpkg haproxy`
2. `echo "LXD_UI_BACKEND_IP=10.201.110.1" > .env.local` if this repo is setup inside a container.
3. `yarnpkg start`
4. Install generated cert or configure SSO to login.

---

Known issues:

1. The Logo component is not display correctly with `yarnpkg start`. Somehow, it looks like the image URL path is correct comparing to existing Incus-UI.
   <img width="242" alt="Screenshot 2024-05-21 at 14 13 04" src="https://github.com/osamuaoki/incus-ui-canonical/assets/1296500/894df152-4cde-40e2-a2f0-13564b486bdf">

2. In [some source codes](https://github.com/osamuaoki/incus-ui-canonical/blob/debian/src/pages/cluster/ClusterList.tsx#L143), it shows "LXD UI", however I see the zabbly package installed is showing "Incus UI". Any magic behind this?
   <img width="535" alt="Screenshot 2024-05-21 at 14 18 15" src="https://github.com/osamuaoki/incus-ui-canonical/assets/1296500/cc72392d-ead4-43fe-82f2-e17ce87f9a55">
